File: http/dotnet/httpclient/src/Middleware/Options/TelemetryHandlerOption.cs
using System;
using System.Net.Http;
using Microsoft.Kiota.Abstractions;

namespace Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options
{
    /// <summary>
    /// The Telemetry request option class
    /// </summary>
    public class TelemetryHandlerOption : IRequestOption
    {
        /// <summary>
        /// A delegate that's called to configure the <see cref="HttpRequestMessage"/> with the appropriate telemetry values.
        /// </summary>
        public Func<HttpRequestMessage, HttpRequestMessage> TelemetryConfigurator { get; set; } = (request) => request;
    }
}

File: http/java/okhttp/lib/src/main/java/com/microsoft/kiota/http/middleware/options/TelemetryHandlerOption.java
package com.microsoft.kiota.http.middleware.options;

import java.util.function.Function;

import com.microsoft.kiota.RequestOption;

import okhttp3.Request;

/**
 * TelemetryHandlerOption class
 */
public class TelemetryHandlerOption implements RequestOption {

    /**
     * A delegate which can be called to configure the Request with desired telemetry values.
     */
    public Function<Request, Request> TelemetryConfigurator = (request) -> request;

}
